test/unit-tests/allocator/context_manager/segment_ctx/segment_ctx_test.cpp

```diff
@@ -924,4 +924,68 @@ TEST(SegmentCtx, ResetSegmentsState_testIfSegmentStateBecomesNVRAMWhenOccupiedSt
     EXPECT_EQ(SegmentState::SSD, segmentInfoData[2].state);
     EXPECT_EQ(SegmentState::FREE, segmentInfoData[3].state);
 }
+
+TEST(SegmentCtx, MoveToFreeState_testIfSegmentFreedWhenStateChanged)
+{
+    NiceMock<MockAllocatorAddressInfo> addrInfo;
+    NiceMock<MockRebuildCtx> rebuildCtx;
+    NiceMock<MockSegmentList> freeSegmentList, ssdSegmentList, rebuildSegmentList, victimSegmentList;
+
+    uint32_t stripesPerSegment = 1024;
+    int numSegments = 1;
+    ON_CALL(addrInfo, GetstripesPerSegment).WillByDefault(Return(stripesPerSegment));
+    ON_CALL(addrInfo, GetnumUserAreaSegments).WillByDefault(Return(numSegments));
+
+    // Given: a segment is VICTIM state and ready to be freed
+    SegmentInfoData* segmentInfoData = new SegmentInfoData[numSegments](0, stripesPerSegment, SegmentState::VICTIM);
+    NiceMock<MockGcCtx> gcCtx;
+    NiceMock<MockTelemetryPublisher> tp;
+    SegmentCtx segmentCtx(&tp, nullptr, segmentInfoData, &rebuildSegmentList, &rebuildCtx, &addrInfo,
+        &gcCtx);
+    segmentCtx.SetSegmentList(SegmentState::FREE, &freeSegmentList);
+    segmentCtx.SetSegmentList(SegmentState::SSD, &ssdSegmentList);
+    segmentCtx.SetSegmentList(SegmentState::VICTIM, &victimSegmentList);
+
+    // Then: The segment should be freed
+    SegmentId segId = 0;
+
+    EXPECT_CALL(victimSegmentList, RemoveFromList(segId)).WillOnce(Return(true));
+    EXPECT_CALL(freeSegmentList, AddToList(segId));
+
+    // When: MoveVictimToFree is called (supposed to be called by GC)
+    bool ret = segmentCtx.MoveToFreeState(segId);
+    EXPECT_EQ(ret, true);
+}
+
+TEST(SegmentCtx, MoveToFreeState_testIfSegmentIsNotFreedWhenStateIsNotChanged)
+{
+    NiceMock<MockAllocatorAddressInfo> addrInfo;
+    NiceMock<MockRebuildCtx> rebuildCtx;
+    NiceMock<MockSegmentList> freeSegmentList, ssdSegmentList, rebuildSegmentList, victimSegmentList;
+
+    uint32_t stripesPerSegment = 1024;
+    int numSegments = 1;
+    ON_CALL(addrInfo, GetstripesPerSegment).WillByDefault(Return(stripesPerSegment));
+    ON_CALL(addrInfo, GetnumUserAreaSegments).WillByDefault(Return(numSegments));
+
+    // Given: a segment is VICTIM state and ready to be freed
```

Comment:
코멘트 한번 확인 부탁드릴께요 혹시 `a segment is SSD state` 로 되어야 할지, 아니면 처음에는 VICTIM이었다가 MoveToFreeState() 실행 중에 SSD 로 다른 쓰레드에 의해 바뀌었기 때문에 `ret`이 false가 되는 경우 (? 뭔가 말로 표현하기가 어렵네요;; ㅎ) 를 표기해야 할까요?? ㅎ
일단 제가 의도를 제대로 파악했다면 `Given: a segment is SSD state and ready to be freed since its valid block count is 0` 라는 코멘트로 변경한다면 맞을까요??

Reply:
헛 맞습니다! 테스트 복붙하다 깜빡했네요 ^^; 수정하겠습니다.

Reply:
코멘트 수정하였습니다! 감사합니다.